src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -137,10 +153,49 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
   closePopover() {
     this.setState({ ...this.state, isPopoverOpen: false });
   }
+  getFilteredDefaultDataSource(defaultDataSource: string | null): string | null {
+    if (!defaultDataSource) return defaultDataSource;
+    const { dataSourceFilter } = this.props;
+    // check if the defaultDataSouce can be filtered out
+    const filteredDataSource = dataSourceFilter
+      ? [{ id: defaultDataSource }].filter(dataSourceFilter)
+      : [{ id: defaultDataSource }];
+    return filteredDataSource.length === 0 ? null : filteredDataSource[0].id;
+  }
+  /**
+   * when call handleSwitchDefaultDatasource, the default data source must exist,
+   * since the button only display when defaultDataSource exist
+   */
+  async handleSwitchDefaultDatasource(defaultDataSourceId: string | null) {
+    /**
+     * default dataSource label must exist when defaultDataSource exist
+     */
+    const defaultDataSourceLabel = (
+      await getDataSourceById(defaultDataSourceId!, this.props.savedObjectsClient!)
+    ).title;
+
+    const defaultDataSourceOption = { id: defaultDataSourceId!, label: defaultDataSourceLabel };
+    // reset the state to close popover and error, selectedOption will be replaced by default option
+    this.setState({
+      selectedOption: [defaultDataSourceOption],
+      showError: false,
+      isPopoverOpen: false,
+    });
+    this.props.onSelectedDataSources!([defaultDataSourceOption]);
```

Comment:
looks like we are enforcing an optional prop here, is it expected? can we add a test that when onSelectedDataSources is not provided, it will not break?

Reply:
Thanks for pointing it out, when tested, the UI not broken, while the console throw error when onSelectedDataSources is not provided, already updated, will add this in test plan